client/asset/doge/doge.go

```diff
@@ -160,32 +192,102 @@ func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network)
 		SingularWallet:           true,
 		UnlockSpends:             true,
 		ConstantDustLimit:        dustLimit,
-		FeeEstimator: func(cl btc.RawRequester, _ uint64) (uint64, error) {
-			confArg, err := json.Marshal(feeConfs)
-			if err != nil {
-				return 0, err
-			}
-			resp, err := cl.RawRequest("estimatefee", []json.RawMessage{confArg})
-			if err != nil {
-				return 0, err
```

Comment:
Can't this FeeEstimator closure stay the same and just call the external API on errors like these?  I'm not seeing the need for `type DogeWallet struct` or why it's composed the way it is.

Reply:
The reason I am not calling it at the FeeEstimator is because I do not have access to the context. 

So I created the DogeWallet struct, so I can call the `Connect` method and initialize the context.